packages/components/src/public/QueryModel/utils.ts

```diff
@@ -126,3 +126,24 @@ export function getQueryModelExportParams(
     };
     return getExportParams(type, schemaQuery, exportOptions, advancedOptions);
 }
+
+export function getSelectRowCountColumnsStr(rawColumns?: string | string[], filterArray?: Filter.IFilter[]): string | string[] {
+    if (filterArray?.length > 0) {
+        const qFilter = filterArray.some(filter => filter.getColumnName() === '*');
+        if (qFilter)
+            return rawColumns;
+    }
+
+    if (!rawColumns || rawColumns === '*') return rawColumns === null ? null : '*';
+
+    const columns: string[] =
+        typeof rawColumns === 'string' ? rawColumns.split(',').map(col => col.trim()) : rawColumns;
+
+    if (columns.length <= 1) return columns[0];
+
+    const columnsLc = columns.map(col => col.toLowerCase());
```

Comment:
as discussed in the Frontend meeting, were you going to look into using `queryInfo.getPkCols()` in some way here instead of hardcoding in the rowid/name cols below?

Reply:
yep, done

Reply:
why not just use pkCols once we get to this part of the function? It is likely going to be a single column but in the case where it is a 2 column PK table, it doesn't hurt anything to include them both. thoughts?

Reply:
Switched to return pkCol instead